src/main/scala/UARTAdapter.scala

```diff
@@ -0,0 +1,156 @@
+package sifive.blocks.devices.uart
+
+import chisel3._
+import chisel3.util._
+import chisel3.core.{IntParam, StringParam}
+
+import freechips.rocketchip.config.{Parameters}
+import freechips.rocketchip.subsystem.{PeripheryBusKey}
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.subsystem.{BaseSubsystem}
+import freechips.rocketchip.config.{Field}
+
+import sifive.blocks.devices.uart._
+import testchipip.{SerialIO}
+
+object UARTAdapterConsts {
+  val DATA_WIDTH = 8
+}
+import UARTAdapterConsts._
+
+case object UARTAdapterKey extends Field[Seq[UARTAdapterParams]](Nil)
+
+case class UARTAdapterParams(
+  baudRateInit: BigInt = BigInt(115200)
+)
+
+class UARTAdapter(uartno: Int)(implicit p: Parameters) extends Module
```

Comment:
Why do we want this adapter rather than the normal sifive-block uart?
Does the sim part also work with that uart?
Assuming we do want this adapter is there a way to reduce the code duplication?

Reply:
Sorry this may not be clear, and Ill add some comments to this. This adapter connects to each SiFive UART in the TestHarness (1 adapter to 1 sifive uart). So for this to work, each configuration needs to have at least 1 SiFive UART.

As for code duplication, this is adapted from the `UARTBridge` made for FireSim... so there was already code duplication there. I think that can be combined/deduped a bit, however, I don't see an obvious way to reduce duplication with the SiFive UART.